### DISCUSSION TITLE
Netbird DNS with docker

### DISCUSSION THREAD

**Discussion by ykorzikowski** [netbirdio/netbird#976] [Q&A / Support]
Hi there, 

I have the following setting: 
- A VM with netbird and docker installed
- A docker container running drone-runner (a worker node for drone ci software) which spawns docker containers running ci jobs

From inside this job container, I want to reach a netbird-peer, which is by network no problem. But I noticed dns issues occuring in a not re-produceable way. Both settings below seems work, until the next or next two days. 

## First Setting

- Leave it as it is -

Adding no additional configuration, as docker uses the host configuration for dns. 

This is working until some point (netbird update, restarting vm, etc.) and then it stops working. 

## Second Setting

- Setting the VMs netbird IP in docker-daemon -

```
#  daemon_file: | Depending on the moon-cycle, this is required or not. Check netbird issue https://github.com/netbirdio/netbird/issues/955
    {
      "dns": ["100.*.*.*"]
    }
```

This is working, but also sometimes this is not working. 

Is there any other recommended way in doing this? I am thinking of using a dedicated dns for docker, because this is a huge trouble maker.

**Comment by ykorzikowski** [comment]
Two identical machines, setup with ansible, result in this: 

```
ansible@vm-drone-runner-1:~$ cat /etc/resolv.conf

nameserver 127.0.0.53
options edns0 trust-ad
search swokiz.net
ansible@vm-drone-runner-1:~$ netbird status
Daemon version: 0.21.4
CLI version: 0.21.4
Management: Connected
Signal: Connected
FQDN: vm-drone-runner-1-1.***.***
NetBird IP: 100.69.***.***/16
Interface type: Kernel
Peers count: 19/22 Connected
ansible@vm-drone-runner-1:~$ sudo docker run alpine ping e.***.***
ping: bad address 'e.***.***'
ansible@vm-drone-runner-1:~$
```

```

ansible@vm-drone-runner-2:~$ cat /etc/resolv.conf
nameserver 127.0.0.53
options edns0 trust-ad
search swokiz.net
ansible@vm-drone-runner-2:~$ netbird status
Daemon version: 0.21.3
CLI version: 0.21.3
Management: Connected
Signal: Disconnected
FQDN: vm-drone-runner-2-2.***.***
NetBird IP: 100.69.***.***/16
Interface type: Kernel
Peers count: 19/22 Connected
ansible@vm-drone-runner-2:~$ sudo docker run alpine ping e.***.***
PING e.***.*** (100.69.***.***): 56 data bytes
64 bytes from 100.69.***.***: seq=0 ttl=63 time=53.569 ms
^C
--- e.***.*** ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 53.569/53.569/53.569 ms
ansible@vm-drone-runner-2:~$
```

**Comment by e-renna** [comment]
The machines might be identical, but one is running netbird 0.21.3 (the working one) and one is running 0.21.4 (the one that fails to resolve). I have been experiencing DNS resolution issues since 0.21.4 (the devs are aware afaik and working on understanding the issue better to find a solution).

Try to rollback netbirs 0.21.3 for the other container, and check if DNS resolution works again
